.clang-format

```diff
@@ -1,5 +1,11 @@
----
+# /.clang-format -*-config-*-
```

Comment:
This will make it slightly harder to keep the .clang-format up to date because clang-format will want to rewrite it when you dump the config. 
I believe it's right now at clang-format-17.
```sh
clang-format-19 --style=file:.clang-format    --dump-config
```
will produce the file that clang-format believes it read, adding the options that were defaulted.

Reply:
> This will make it slightly harder to keep the .clang-format up to date because clang-format will want to rewrite it when you dump the config.

I'm OK to remove this file, but wouldn't be consistent for other linters? I think I would like to change it manually (using --dump-config into a temporary file). What do you think?

btw, I tried with clang 17/18/19, and I cannot get the current config. Would you like to update it with clang-format-19 in this PR? (The required changes will be applied in later commits or in a new PR).

Reply:
It's the comment itself that gets lost. `clang-format` configs are generally one way upgradable but otherwise stable. That is, clang-format-N+1 will read an N file and default the elements that are added, so that when you run dump-config you get a file that only cf-N+1 understands but that produces the same output as the old config. 

Caveats around syntax that clang-format didn't used to understand, of course. 

We don't need to be eager about upgrading unless we decide we need the new knobs. 

I've left myself a TODO to see if clang-format changes matter for the few files we have, and if so, we probably don't need to upgrade. 

It's just that the comments are easy to get lost, and I wouldn't want anyone to read too much into that.